### PR TITLE
OBSDOCS-1008 - 5.9.1 Release Notes

### PR DESCRIPTION
**Pull request by shreyasiddhartha**
Change type: Doc update; Logging Z-Stream Release Notes - 5.9.1
Doc JIRA: https://issues.redhat.com/browse/OBSDOCS-1008

Fix Version: 4.13+

Doc Preview: https://75242--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/logging/logging_release_notes/logging-5-9-release-notes.html#logging-release-notes-5-9-1_logging-5-9-release-notes 

SME Review: @periklis 
QE Review: @kabirbhartiRH 
Peer Review: @mburke5678

### CODE REVIEW

**Comment by kabirbhartiRH** [comment]
Couple of bugs which are missing and not having release notes
https://issues.redhat.com/browse/LOG-4672
https://issues.redhat.com/browse/LOG-5062

**Review by periklis — CHANGES_REQUESTED** [review]
Missing bug fixes: LOG-4672, LOG-5062, LOG-5307, LOG-5309

**Review by periklis — APPROVED** [review]
LGTM

**Comment by kabirbhartiRH** [comment]
And here too, [LOG-5401](https://issues.redhat.com/browse/LOG-5401) and [LOG-5395](https://issues.redhat.com/browse/LOG-5395) are enhancements, not bugs.

File: modules/logging-release-notes-5-9-1.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+* Before this update, a drop filter without a defined `dropTestsSpec` would cause a segfault. With this update, there is a validation error if a prune filter is without a defined `puneFilterSpec`. (link:https://issues.redhat.com/browse/LOG-5323[LOG-5323])
+
+* Before this update, the {loki-op} did not validate the S3 endpoint used in the storage secret. With this update, the validation process ensures the S3 endpoint is a valid S3 URL, and the `LokiStack` status updates to indicate any invalid URLs. (link:https://issues.redhat.com/browse/LOG-5395[LOG-5395])
+
+* Before this update, the {loki-op} did not validate the correctness of the S3 endpoint URL format used in the storage secret. With this update, the S3 endpoint URL goes through a validation step that reflects on the status of the `LokiStack`. (link:https://issues.redhat.com/browse/LOG-5397[LOG-5397])
+
+* Before this update, {loki-op} set up Loki to use path-based style access for AWS S3, which has been deprecated. With this update, it defaults to virtual-host style without users needing to change their configuration. (link:https://issues.redhat.com/browse/LOG-5401[LOG-5401])
```
Comment (mburke5678) [inline modules/logging-release-notes-5-9-1.adoc:20]:
```suggestion
* Before this update, the {loki-op} configured Loki to use path-based style access for AWS S3, which has been deprecated. With this update, the {loki-op} defaults to virtual-host style without users needing to change their configuration. (link:https://issues.redhat.com/browse/LOG-5401[LOG-5401])
```

File: modules/logging-release-notes-5-9-1.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+* Before this update, poorly formatted timestamp fields in audit log records led to `WARN` messages in {clo} logs. With this update, a remap transformation ensures that the timestamp field will be properly formatted. (link:https://issues.redhat.com/browse/LOG-4672[LOG-4672])
```
Comment (mburke5678) [inline modules/logging-release-notes-5-9-1.adoc:22]:
```suggestion
* Before this update, poorly formatted timestamp fields in audit log records led to `WARN` messages in {clo} logs. With this update, a remap transformation ensures that the timestamp field is properly formatted. (link:https://issues.redhat.com/browse/LOG-4672[LOG-4672])
```

File: modules/logging-release-notes-5-9-1.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+* Before this update, the validation feature for {logging} didn't align with the current documentation. With this update, the validation feature for {logging} is aligned with the documentation. Now, during validation, the system checks if a `ClusterLogForwarder` with the same name exists in the same namespace. If not, it throws an exception. (link:https://issues.redhat.com/browse/LOG-5062[LOG-5062])
```
Comment (mburke5678) [inline modules/logging-release-notes-5-9-1.adoc:24]: > did not align with the current documentation
This doesn't seem right. The code doesn't align with the docs; the doc align with the code. I am not sure how to correct this, as I don't fully understand the issue. 

Also, the bug Jira is `Unresolved`. Should this one be documented?

File: modules/logging-release-notes-5-9-1.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+* Before this update, {loki-op} set up Loki to use path-based style access for AWS S3, which has been deprecated. With this update, it defaults to virtual-host style without users needing to change their configuration. (link:https://issues.redhat.com/browse/LOG-5401[LOG-5401])
+
+* Before this update, the validation feature for output config required a TLS URL, even for services like Amazon CloudWatch or Google Cloud Logging where a URL is not needed by design. With this update, the validation logic for services without URLs received improvement, and the error message became more informative. (link:https://issues.redhat.com/browse/LOG-5307[LOG-5307])
```
Comment (mburke5678) [inline modules/logging-release-notes-5-9-1.adoc:28]:
```suggestion
* Before this update, the validation feature for output config required a TLS URL, even for services such as Amazon CloudWatch or Google Cloud Logging, where a URL is not needed by design. With this update, the validation logic for services without URLs are improved and the error message are more informative. (link:https://issues.redhat.com/browse/LOG-5307[LOG-5307])
```

**Comment by mburke5678** [comment]
@shreyasiddhartha A few suggestions. Otherwise LGTM

File: modules/logging-release-notes-5-9-1.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+* Before this update, poorly formatted timestamp fields in audit log records led to `WARN` messages in {clo} logs. With this update, a remap transformation ensures that the timestamp field will be properly formatted. (link:https://issues.redhat.com/browse/LOG-4672[LOG-4672])
+
+* Before this update, the validation feature for {logging} didn't align with the current documentation. With this update, the validation feature for {logging} is aligned with the documentation. Now, during validation, the system checks if a `ClusterLogForwarder` with the same name exists in the same namespace. If not, it throws an exception. (link:https://issues.redhat.com/browse/LOG-5062[LOG-5062])
+
+* Before this update, {loki-op} set up Loki to use path-based style access for AWS S3, which has been deprecated. With this update, it defaults to virtual-host style without users needing to change their configuration. (link:https://issues.redhat.com/browse/LOG-5401[LOG-5401])
```
[the remark this answers is not in the document]
Comment (shreyasiddhartha) [reply modules/logging-release-notes-5-9-1.adoc:26]: Remove the second entry.

File: modules/logging-release-notes-5-9-1.adoc (hunk 3 above)
Comment (shreyasiddhartha) [reply modules/logging-release-notes-5-9-1.adoc:24]: It's marked as `Done-Errata` now. So documented and rephrased to make more sense. Thanks!

**Review by opayne1 — COMMENTED** [review]
lgtm

**Comment by opayne1** [comment]
@shreyasiddhartha You will need to do a manual CP if this needs to go in 4.12

**Comment by shreyasiddhartha** [comment]
> @shreyasiddhartha You will need to do a manual CP if this needs to go in 4.12

There was some confusion regarding versioning. It should be 4.13+. Thanks!

**Pull request merged**